### ISSUE TITLE
"Text contrast" [afw4f7, 09o5cg]: Should rule apply to `X` for close?

### ISSUE THREAD

**Issue by Jym77** [auto-wcag/auto-wcag#2132]
Consider:
```html
<style>
  .bad { color: #AAA; background-color: white }
</style>
<button class="bad" aria-label="Previous">&lt</button>
<button class="bad" aria-label="Close">X</button>
<button class="bad" aria-label="Next">&gt</button>
```
* It seems obvious that these buttons do need good contrast and be readable by users with low vision.
* OTOH, it also seems that the "text" is actually non-text content and should therefore pass the rule(s).
* I think this should also pass 1.4.3/1.4.6 as not being text, but likely should fail ~1.4.1 as conveying the info only by color (hue, not contrast)~ 1.4.11.

I would say that therefore the rule should pass these, and I am rather in favour of adding an explicit example of it to make it clear. The current example of "non-text content" we have is fairly contrived (`----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@`) and not very realistic.

**Comment by dan-tripp-siteimprove** [comment]
This overlaps with our [recent alfa conversation](https://github.com/Siteimprove/alfa/issues/1166) .

The way I see it, your example of "X for close" is clearly in the "non-text == true and decorative == false" category, and so would make a helpful example.   (A "passed" example, yes.) 

For the current passed example of "----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@" it's unclear whether that's in the "non-text == true and decorative == false" category, or the "non-text == true and decorative == true" category.  So it's not helpful, and should be removed.

**Comment by Jym77** [comment]
Update the "non text content" examples with something more "real world".

**Comment by dan-tripp-siteimprove** [comment]
What do you think of replacing this:
~~~
#### Passed Example 7

This text does not convey anything in human language.

```html
<p style="color: #000; background: #666;">
	----=====++++++++___________***********%%%%%%%%%%%±±±±@@@@@@@@
</p>
```
~~~
... with this? 
~~~
#### Passed Example 7

This text does not convey anything in human language, so it passes the rule.  It also passes [Success Criterion 1.4.3: Contrast (Minimum)][sc143], because it is not (text)[https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html#dfn-text].  It fails [Success Criterion 1.4.1: Use of Color][sc141].

```html
<button style="color: #AAA; background-color: white" aria-label="Close">X</button>
```
~~~
... and I would add this line to the bottom of the page:
~~~
[sc141]: https://www.w3.org/TR/WCAG21/#use-of-color 'WCAG 2.1, Success Criterion 1.4.1: Use of Color'
~~~

**Comment by giacomo-petri** [comment]
I disagree with the inclusion of 1.4.1 in this example.

The failure of 1.4.1 is unwarranted because the color of the "X" close button—whether green, blue, red, or black—does not impact the comprehensibility of the element. The element's purpose is determined by its shape, not its color. While it holds true for instances where different colors of "X" denote availability/non-availability (e.g., in a comparison table, green and red "X"), the current situation is distinctly different.

In this particular case, the only failure is in 1.4.11, relating to non-text contrast.
